logstreams/src/main/java/io/zeebe/logstreams/impl/LogStorageAppender.java

```diff
@@ -181,7 +198,35 @@ public String getName() {
 
   @Override
   protected void onActorStarting() {
+    final CompletableActorFuture<Long> appendIndexFuture = new CompletableActorFuture<>();
+    distributedLog
+        .getLastAppendIndex()
+        .thenAccept(appendIndexFuture::complete)
+        .exceptionally(
+            t -> {
+              appendIndexFuture.completeExceptionally(t);
+              return null;
+            });
+
+    actor.runOnCompletionBlockingCurrentPhase(
```

Comment:
This means, `onActorStarted()` is called only after this callback is executed. Right?

Reply:
yes exactly